tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_WenoHelpers.cpp

```diff
@@ -82,11 +97,215 @@ void test_oscillation_indicator_3d() noexcept {
   CHECK(indicator == approx(expected));
 }
 
+struct ScalarTag : db::SimpleTag {
+  using type = Scalar<DataVector>;
+  static std::string name() noexcept { return "Scalar"; }
+};
+
+template <size_t VolumeDim>
+struct VectorTag : db::SimpleTag {
+  using type = tnsr::I<DataVector, VolumeDim>;
+  static std::string name() noexcept { return "Vector"; }
+};
+
+void test_reconstruction_1d() noexcept {
+  const double neighbor_linear_weight = 0.001;
+  const auto mesh = Mesh<1>({{5}}, Spectral::Basis::Legendre,
+                            Spectral::Quadrature::GaussLobatto);
+  const auto coords = logical_coordinates(mesh);
+
+  const auto evaluate = [&coords](
+      const std::array<double, 5>& coeffs) noexcept {
+    const auto& x = get<0>(coords);
+    return DataVector{coeffs[0] + coeffs[1] * x + coeffs[2] * square(x) +
+                      coeffs[3] * cube(x) + coeffs[4] * pow<4>(x)};
+  };
+
+  auto scalar = ScalarTag::type{{{evaluate({{1., 2., 0., 0.5, 0.1}})}}};
+
+  std::unordered_map<std::pair<Direction<1>, ElementId<1>>,
+                     Variables<tmpl::list<ScalarTag>>,
+                     boost::hash<std::pair<Direction<1>, ElementId<1>>>>
+      neighbor_vars{};
+  auto& vars_lower_xi =
+      neighbor_vars[std::make_pair(Direction<1>::lower_xi(), ElementId<1>(1))];
+  auto& vars_upper_xi =
+      neighbor_vars[std::make_pair(Direction<1>::upper_xi(), ElementId<1>(2))];
+  vars_lower_xi.initialize(mesh.number_of_grid_points());
+  vars_upper_xi.initialize(mesh.number_of_grid_points());
+  get<ScalarTag>(vars_lower_xi) =
+      ScalarTag::type{{{evaluate({{0., 1., 0., 1., 0.}})}}};
+  get<ScalarTag>(vars_upper_xi) =
+      ScalarTag::type{{{evaluate({{0., 0., 1., 1., 2.}})}}};
+
+  SlopeLimiters::Weno_detail::reconstruct_from_weighted_sum<ScalarTag>(
+      make_not_null(&scalar), mesh, neighbor_linear_weight, neighbor_vars);
+
+  // Expected result computed in Mathematica by computing oscillation indicator
```

Comment:
Why not do this with calling into python rather than some Mathematica code that I can't check to see if you did it correctly and isn't in the repo? One of the main goals of having `pypp` was to support computing the second result inside the code rather than in something proprietary like Mathematica.

Reply:
Will think about this one... it is not obvious to me if it is reasonably straightforward to compute spectral derivatives and transformations in our Python env, or alternatively, if it is reasonably straightforward to pass the sets of extra arguments though `pypp` in the first place.

Reply:
It was not straightforward to test this in Python, for several reasons:
- the complicated datastructure associated with input data (a map of tensors)
- either the oscillation indicator needs to be computed in Python, which requires spectral transforms or differentiations, or the indicator for each tensor component for each neighbor needs to be passed in as an extra argument (which isn't supported).